A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js

```diff
@@ -318,19 +326,22 @@ class A320_Neo_EICAS extends Airliners.BaseEICAS {
             this.SwitchToPageName(this.LOWER_SCREEN_GROUP_NAME, this.pageNameWhenUnselected);
         }
 
-        // ECAM all button
-        this.ecamAllButtonState = SimVar.GetSimVarValue("L:A32NX_ECAM_ALL_Push_IsDown", "Bool");
+        if (ecamAllButtonState) { // button press
+
+            if (!this.ecamAllButtonTimerStarted) {
+                this.changePage(this.lowerScreenPages[(this.currentPage + 1) % this.lowerScreenPages.length].name);
+                this.ecamAllButtonTimerStarted = true;
 
-        if (this.ecamAllButtonState && !this.ecamAllButtonPrevState) { // button press
-            this.changePage(this.lowerScreenPages[(this.currentPage + 1) % this.lowerScreenPages.length].name);
-            this.ecamCycleInterval = setInterval(() => {
+            } else if (this.ecamAllButtonTimer <= 0) {
                 this.changePage(this.lowerScreenPages[(this.currentPage + 1) % this.lowerScreenPages.length].name);
-            }, 1000);
+                this.ecamAllButtonTimer = 1000;
+            }
         } else if (!this.ecamAllButtonState && this.ecamAllButtonPrevState) { // button release
-            clearInterval(this.ecamCycleInterval);
+            this.ecamAllButtonTimer = 1000;
+            this.ecamAllButtonTimerStarted = false;
         }
 
-        this.ecamAllButtonPrevState = this.ecamAllButtonState;
+        this.ecamAllButtonPrevState = ecamAllButtonState;
```

Comment:
**question**: Does delta time differ from the time "seen" by `setInterval` and is that then the reason to write your own `setInterval`-like implementation?
**suggestion**: If no: use `setInterval`, if yes: extract `DeltaTimeIntervalFn` out into its own type to reduce the "going up and down abstraction levels" here.